### PR TITLE
[ProgressV2] fix sort by name not actually changing the table

### PR DESCRIPTION
**Pull request by lfryemason**
https://github.com/code-dot-org/code-dot-org/assets/25193259/c6b846c3-18d7-4cbe-a508-79effed4a06f

## Links
[TEACH-889](https://codedotorg.atlassian.net/browse/TEACH-889)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by etaderhold — APPROVED** [review]
Huh, weird that this makes all the difference. A test to verify this behavior would be nice.

**Comment by lfryemason** [comment]
@etaderhold I made enough changes to the tests that I think I want a second review. I migrated the tests to RTL and added a couple test cases to make sure the table is sorted correctly too

**Review by etaderhold — APPROVED** [review]

**Pull request merged**
